test.js

```diff
@@ -0,0 +1,18 @@
+var assert = require('assert');
+var spawnSync = require('child_process').spawnSync;
+var greeting = require('./name');
+var should = require('chai').should();
+
+describe('test to check if the arguments are being processed', function(){
+  it('should pass a argument on the command line', function(){
+    var greet = spawnSync('node', ['greet.js', 'max']);
+    var greetString = greet.stdout.toString();
+    assert.deepEqual(greetString.slice(0,5), 'hello','did not get value from command line');
+  });
+  it('should not start with a number', function(){
+    var greet = spawnSync('node', ['greet.js', 'foo']);
+    var greetString = greet.stdout.toString();
+    var notNumber = isNaN(greetString.slice(0,1));
+    notNumber.should.equal(true);
+  });
```

Comment:
well originally I had '3' instead of 'foo' and it was not passing in the
number test so I was just making sure that if I entered a number it would
not say hello back so the first test I was just seeing if I could pass a
argument. Then I built on that in the second test to make sure a number
wasn't a valid name. I did it just for practice, I know that the assignment
didn't ask for that. Does that make sense or did I just make it more
confusing?

On Thu, Jan 7, 2016 at 1:08 PM, Marty Nelson <notifications@github.com>
wrote:

> In test.js
> <https://github.com/max33nau/testModularPatterns/pull/2#discussion_r49125895>
> :
>
> > +var spawnSync = require('child_process').spawnSync;
> > +var greeting = require('./name');
> > +var should = require('chai').should();
> > +
> > +describe('test to check if the arguments are being processed', function(){
> > +  it('should pass a argument on the command line', function(){
> > +    var greet = spawnSync('node', ['greet.js', 'max']);
> > +    var greetString = greet.stdout.toString();
> > +    assert.deepEqual(greetString.slice(0,5), 'hello','did not get value from command line');
> > +  });
> > +  it('should not start with a number', function(){
> > +    var greet = spawnSync('node', ['greet.js', 'foo']);
> > +    var greetString = greet.stdout.toString();
> > +    var notNumber = isNaN(greetString.slice(0,1));
> > +    notNumber.should.equal(true);
> > +  });
>
> not sure what you're going for here. You're not testing an input that
> *does* have a number, so it seems like all your testing beyond the first
> test is whether javascript string.slice and isNaN work.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/max33nau/testModularPatterns/pull/2/files#r49125895>.
>